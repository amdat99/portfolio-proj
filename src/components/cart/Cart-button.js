import React from 'react';
import { connect } from 'react-redux';

import {toggleCartHidden}  from '../../redux/cart/cart.actions'
import { selectCartItemsCount } from '../../redux/cart/cart.selectors'
import { createStructuredSelector} from 'reselect'


import './Cart.scss'

export function CartButton({toggleCartHidden, itemCount}) {
    return (
        <div className = 'cart-button-container'   onClick={toggleCartHidden} 
        >   
            <span> CART </span>
            <span className = 'cart-button-icon'>ðŸ’°</span>
           
            <span className = 'cart-button-count'>{itemCount}</span>
            
        </div>
    );
}

const mapDispatchToProps = (dispatch) => ({
    toggleCartHidden: () => dispatch(toggleCartHidden()),
})

const mapStateToProps = (createStructuredSelector)  ({
    itemCount: selectCartItemsCount
});

export default connect(mapStateToProps, mapDispatchToProps)(CartButton) 