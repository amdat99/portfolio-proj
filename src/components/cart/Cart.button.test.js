import React from 'react';

import { shallow} from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import toJson from 'enzyme-to-json';


import {CartButton} from './Cart-button'

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
// describe('cart item', () => {
      
        let mocktoggleCartHidden = jest.fn();
        let wrapper
        let wrapper2

        
  beforeEach(() => {

   const mockProps = {
            itemCount: '1',
            toggleCartHidden: mocktoggleCartHidden,
          
          }; 
          
           wrapper = toJson(shallow(<CartButton  {...mockProps} />))
           wrapper2 = shallow(<CartButton {...mockProps} />);
        })
        



it('expect to render cartButton component', () => {
    expect(wrapper).toMatchSnapshot();
})

it('expect togglecart hidden to be called on click', () => {
    wrapper2.find('.cart-button-container').simulate('click');
    expect(mocktoggleCartHidden).toHaveBeenCalled()
})

it('the count number should be rendered', () => {
    const itemCount = parseInt(wrapper2.find('.cart-button-count').text());
    expect(itemCount).toBe(1);
  });



