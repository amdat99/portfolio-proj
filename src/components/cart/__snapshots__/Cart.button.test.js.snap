// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render cartButton component 1`] = `
<div
  className="cart-button-container"
  onClick={[MockFunction]}
>
  <span>
     CART 
  </span>
  <span
    className="cart-button-icon"
  >
    ðŸ’°
  </span>
  <span
    className="cart-button-count"
  >
    1
  </span>
</div>
`;
